src/components/themed/WalletListCurrencyRow.js

```diff
@@ -38,166 +24,56 @@ type Props = {
-  // Return zero result
-  if (zeroString(differencePercentage)) return result('0.00')
-  // If not zero, create the `percentageString`
-  const percentageString = abs(differencePercentage)
-  // Return Positive result if greater then zero
-  if (gt(differencePercentage, '0')) return result(`+${percentageString}`, 'positive')
-  // If it's not zero or positive, it must be a Negative result
-  return result(`-${percentageString}`, 'negative')
-}
-
+/**
+ * Renders a WalletListRow with the hide-able balance children.
+ */
 export const WalletListCurrencyRowComponent = (props: Props) => {
   const { currencyCode, showRate = false, onPress, onLongPress, gradient, walletId, walletName, tokenCode } = props
   const dispatch = useDispatch()
   const theme = useTheme()
   const styles = getStyles(theme)
 
-  const edgeWallet = useSelector(state => state.core.account.currencyWallets[walletId])
-  const exchangeRates = useSelector(state => state.exchangeRates)
+  const currencyInfo = useSelector(state => state.core.account.currencyWallets[walletId].currencyInfo)
+  const fiatCurrencyCode = useSelector(state => state.core.account.currencyWallets[walletId].fiatCurrencyCode)
+  const balance = useSelector(state => state.core.account.currencyWallets[walletId].balances[currencyCode] ?? '0')
+  const name = useSelector(state => state.core.account.currencyWallets[walletId].name)
   const showBalance = useSelector(state => state.ui.settings.isAccountBalanceVisible)
 
-  // Subscribe to wallet updates
-  const { currencyInfo } = edgeWallet
-  const [fiatCurrencyCode, setFiatCurrencyCode] = useState(edgeWallet.fiatCurrencyCode)
-  const [balances, setBalances] = useState(edgeWallet.balances)
-  const [name, setName] = useState(edgeWallet.name)
-
-  useEffect(() => {
-    const watchers = []
-    if (edgeWallet != null) {
-      watchers.push(edgeWallet.watch('fiatCurrencyCode', setFiatCurrencyCode))
-      watchers.push(edgeWallet.watch('name', setName))
-      watchers.push(edgeWallet.watch('balances', setBalances))
-    }
-    return () => watchers.forEach(cleaner => cleaner())
-  }, [edgeWallet])
-
   // Crypto Amount And Exchange Rate
-  const balance = balances[currencyCode] ?? '0'
-  const denomination = dispatch(getDisplayDenominationFromState(currencyInfo.pluginId, currencyCode))
   const exchangeDenomination = dispatch(getExchangeDenominationFromState(currencyInfo.pluginId, currencyCode))
-  const fiatDenomination = getDenomFromIsoCode(fiatCurrencyCode)
   const isoFiatCurrencyCode = fixFiatCurrencyCode(fiatCurrencyCode)
-  const rateKey = `${currencyCode}_${isoFiatCurrencyCode}`
-  const exchangeRate = !zeroString(exchangeRates[rateKey]) ? exchangeRates[rateKey] : '1'
   const cryptoExchangeMultiplier = exchangeDenomination.multiplier
 
-  const nativeCryptoAmount = getCryptoAmount({ balance, exchangeRate, exchangeDenomination, fiatDenomination, denomination, currencyCode })
-
   const { fiatText: fiatBalanceText } = useFiatText({
```

Comment:
We were supposed to switch to `<FiatText>` here, rather than using a hook.

Actually, can we not modify the wallet list rows here at this point? I have another PR, #3336, that goes pretty deep into refactoring the wallet list rows, and these two PR's are conflicting. Once we get the new text components done, as well as the refactors to the wallet list, we can do a third PR that brings both together.